### PR TITLE
Properly placed pages tests

### PR DESCRIPTION
**Pull request by planarvoid**
I'm using injected `uiContext` and `commonPoolContext` in view models in `SearchListViewModel` and `PagesViewModel`. I've separated the tests for these two to become truly unit tests. I've removed the `_pageMap` which was unnecessary because you can keep the value in `pageMap`. I've made it immutable because immutable collections are much easier to maintain and much less error-prone. I've added some test coverage but I feel like there are too many changes in this PR already so I'll add more in a separate PR.  

I'm removing `LiveDataUtils` because it's not necessary any more since we're using synchronous `liveData.setValue` instead of async `liveData.postValue`.

To test: Go through the new Pages activity and check that everything still works as expected.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt [hunk 1]
```diff
@@ -274,27 +270,25 @@ class PagesViewModel
...
                     pageStore.uploadPageToServer(page)
                 }
             }
         }
         action.undo = {
-            launch(CommonPool) {
+            launch(commonPoolContext) {
                 pageMap[page.remoteId]?.let { changed ->
-                    changed.parent = _pageMap[oldParent]
-                    pageMap = _pageMap
```
Comment (0nko) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt]: Same here, lost instant updates.

File: WordPress/src/test/java/org/wordpress/android/viewmodel/pages/PagesViewModelTest.kt [hunk 2]
```diff
@@ -34,42 +43,125 @@ class PagesViewModelTest {
...
 
-        val listStates = listStateObserver.awaitValues(2)
-
         assertThat(listStates).containsExactly(FETCHING, DONE)
-        refreshPagesObserver.awaitNullableValues(2)
+        assertThat(pages).hasSize(2)
+    }
+
+    @Test
+    fun onPageEditFinishedReloadSite() = runBlocking<Unit> {
```
Comment (0nko) [inline WordPress/src/test/java/org/wordpress/android/viewmodel/pages/PagesViewModelTest.kt]: Are you planning to add the `test()` function as you did in FluxC?

File: WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt [hunk 3]
```diff
@@ -274,27 +270,25 @@ class PagesViewModel
         val oldParent = page.parent?.remoteId ?: 0
 
         val action = PageAction(UPLOAD) {
-            launch(CommonPool) {
+            launch(commonPoolContext) {
                 if (page.parent?.remoteId != parentId) {
-                    page.parent = _pageMap[parentId]
-                    pageMap = _pageMap
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt]: you're right! I missed that. I'll call the setter explicitly 👍

[the remark this answers is not in the document]
Comment (0nko) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt]: Perfect!

**Comment by planarvoid** [comment]
@0nko Everything should be fixed now 👍

**Comment by 0nko** [comment]
Looking great! :shipit:

**Pull request merged**